test/lib/util/reader-util.test.js

```diff
@@ -1,5 +1,56 @@
-const { treatUrl, treatMapping } = require("../../../src/lib/util/reader-util");
-global.console = { error: jest.fn() };
+const {
+  treatUrl,
+  treatMapping,
+  executeUrlExpressions
+} = require("../../../src/lib/util/reader-util");
+const { logger } = require("../../../src/lib/common");
+jest.mock("../../../src/lib/common");
+
+test("executeUrlExpressions without expressions", () => {
+  // Arrange
+  const string =
+    "https://raw.githubusercontent.com/kiegroup/droolsjbpm-build-bootstrap/main/.ci/pull-request-config.yaml";
+
+  // Act
+  const result = executeUrlExpressions(string);
+
+  // Assert
+  expect(result).toBe(string);
+});
+
+test("executeUrlExpressions with 1 expression", () => {
+  // Arrange
+  process.env = Object.assign(process.env, {
+    GITHUB_BASE_REF: "1.2.3"
+  });
+  const string =
+    "https://raw.githubusercontent.com/kiegroup/droolsjbpm-build-bootstrap/%{process.env.GITHUB_BASE_REF.replace(/(\\d*)\\.(.*)\\.(.*)/g, (m, n1, n2, n3) => `${+n1+7}.${n2}.${n3}`)}/.ci/pull-request-config.yml";
+
+  // Act
+  const result = executeUrlExpressions(string);
+
+  // Assert
+  expect(result).toBe(
+    "https://raw.githubusercontent.com/kiegroup/droolsjbpm-build-bootstrap/8.2.3/.ci/pull-request-config.yml"
+  );
+});
+
+test("executeUrlExpressions with 2 expressions", () => {
```

Comment:
Could you also test with an expression and no matching of version ?

for example with
```
process.env = Object.assign(process.env, {
    GITHUB_BASE_REF: "1.2.3"
  });
```
and
```
const string =
    "https://raw.githubusercontent.com/kiegroup/droolsjbpm-build-bootstrap/%{process.env.GITHUB_BASE_REF.replace(/(\\d*)\\.(.*)\\.(.*)/g, (m, n1, n2, n3) => `${+n1+7}.${n2}.${n3}`)}/.ci/pull-request-config.yml";
```
This should give "https://raw.githubusercontent.com/kiegroup/droolsjbpm-build-bootstrap/main/.ci/pull-request-config.yml"

Reply:
@radtriste good catch. I guess this is the test you mean https://github.com/kiegroup/build-chain-configuration-reader/pull/23/commits/63acef64c83fba71d82fc2c4e0ea5424a4294aec

Reply:
yes

Reply:
thanks man!